src/services/polling.service.ts

```diff
@@ -0,0 +1,201 @@
+    this.active = false;
+    this.currentPollingIntervalMs = this.pollingIntervalMs;
+  }
+
+  private async getDbRecentTransactionsSorted(
+    minLastUpdated: Date,
+  ): Promise<
+    Pick<Transaction, "id" | "createdAt" | "lastUpdated" | "status">[]
+  > {
+    return await Transaction.find({
+      select: ["id", "createdAt", "lastUpdated", "status"],
+      where: {
+        lastUpdated: MoreThan(minLastUpdated),
+      },
+      order: { createdAt: "ASC" },
+    });
+  }
+
+  private hoursAgoDate(hours: number) {
+    const date: Date = new Date();
+    date.setHours(date.getHours() - hours);
+    return date;
+  }
+
+  private isCompletedTx(tx: { status: TransactionStatus }) {
+    return PollingService.FINAL_STATUSES.includes(tx.status);
+  }
+
+  private txResponseToTxDetails(tr: TransactionResponse): ITransactionDetails {
+    return {
+      ...tr,
+      subStatus:
+        TransactionSubStatus[tr.subStatus as keyof typeof TransactionSubStatus],
+      destinations: tr.destinations?.map((dest) => ({
+        ...dest,
+        amountUSD: Number(dest.amountUSD),
+      })),
+      signedMessages: tr.signedMessages!,
+    };
+  }
+
+  private async pollAndUpdateAll() {
+    console.log("PollingService: pollAndUpdateAll");
+    try {
+      // Get transactions from Fireblocks:
+      const txsResponses: TransactionResponse[] = await fetchAllTxs(
+        (pagePath) =>
+          this.clients.admin.getTransactionsWithPageInfo({}, pagePath),
+      );
+      for (const txResponse of txsResponses) {
+        const tx = this.txResponseToTxDetails(txResponse);
+        const { id, status } = tx;
+        await patchTransactionAmountUsd(tx, this.clients.cmc);
+        await handleTransactionCreatedOrUpdated(id, status, tx);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Increases the current polling frequency for a short period of time
+   */
+  private increasePollingFrequency() {
+    console.log(
+      "PollingService: increasePollingFrequency - increase frequency",
+    );
+    this.currentPollingIntervalMs = this.pollingShortIntervalMs;
+
+    if (this.incomingTxTimer) {
+      clearTimeout(this.incomingTxTimer);
+      this.incomingTxTimer = null;
+    }
+
+    this.incomingTxTimer = setTimeout(() => {
+      console.log(
+        "PollingService: increasePollingFrequency - reset back to normal",
+      );
+      this.currentPollingIntervalMs = this.pollingIntervalMs;
+    }, 120_000);
+  }
+
+  /**
+   * Polls for recent transactions from FB BE and updates their status in DB.
+   */
+  async pollAndUpdate() {
+    try {
+      // Get recent transactions from DB (ignore transactions that were not updated for more than 48 hours):
+      const dbRecentTxs = await this.getDbRecentTransactionsSorted(
+        this.hoursAgoDate(48),
+      );
+      // Get createdAt of earliest non-completed tx
+      const firstNonCompletedTx = dbRecentTxs.find(
+        (tx) => !this.isCompletedTx(tx),
+      );
```

Comment:
we can add this condition to the query to let SQL handle it

Reply:
when calling FB we cannot filter by statuses (only a single status filter is supported),
and since we need to be able to know which transactions do not exist in DB yet,
we need to get from DB all txs, even "completed" ones.

So this logic can't be moved to the SQL